File: benchmark/c/cpw/number_digits_product_two_numbers/number_digits_product_two_numbers.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int number_digits_product_two_numbers ( int a, int b ) ;
int number_digits_product_two_numbers ( int a, int b ) {
  int count = 0;
  int p = abs ( a * b );
  if ( p == 0 ) return 1;
  while ( p > 0 ) {
    count ++;
    p = p / 10;
  }
  return count;
}

File: translation_result/else_bms/fail/c/cpw/number_digits_product_two_numbers/gemini/Hinted/number_digits_product_two_numbers.rs

fn number_digits_product_two_numbers(a: i32, b: i32) -> i32 {
    let p = (a * b).abs();
    if p == 0 {
        return 1;
    }
    let mut count = 0;
    let mut p = p;
    while p > 0 {
        count += 1;
        p /= 10;
    }
    count
}
